server/channels/app/post.go

```diff
@@ -1447,7 +1447,12 @@ func (a *App) parseAndFetchChannelIdByNameFromInFilter(c request.CTX, channelNam
 		if err != nil {
 			return nil, err
 		}
-		return channel, nil
+		if (channel != nil) {
+			return channel, nil
+		}
+	}
+	if strings.HasPrefix(channelName, "~") || strings.HasPrefix(channelName, "@") {
+		channelName = channelName[1:]
```

Comment:
I don't think it makes sense to allow channels prefixed by `@`, so we can leave the `return channel, nil` as it was and then, either simply check by `strings.HasPrefix(channelName, "~")`, or do `channelName = strings.TrimPrefix(channelName, "@")` as you do with users below.

Reply:
Yeah that's reasonable. I've changed it now!